incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -294,142 +290,117 @@ func (r *HierarchyConfigReconciler) syncParent(log logr.Logger, inst *api.Hierar
 	}
 }
 
-func (r *HierarchyConfigReconciler) syncLabel(log logr.Logger, nsInst *corev1.Namespace, ns *forest.Namespace) {
-	// Depth label only makes sense if there's no error condition.
-	if ns.HasCritCondition() {
-		return
-	}
-
-	// Pre-define label depth suffix
-	labelDepthSuffix := fmt.Sprintf(".tree.%s/depth", api.MetaGroup)
-
-	// Remove all existing depth labels.
-	for k := range nsInst.Labels {
-		if strings.HasSuffix(k, labelDepthSuffix) {
-			delete(nsInst.Labels, k)
-		}
-	}
-
-	// AncestryNames includes the namespace itself.
-	ancestors := ns.AncestryNames(nil)
-	for i, ancestor := range ancestors {
-		l := ancestor + labelDepthSuffix
-		dist := strconv.Itoa(len(ancestors) - i - 1)
-		metadata.SetLabel(nsInst, l, dist)
-	}
-}
-
 // syncChildren looks at the current list of children and compares it to the children that
 // have been marked as required. If any required children are missing, we add them to the in-memory
 // forest and enqueue the (missing) child for reconciliation; we also handle various error cases.
-func (r *HierarchyConfigReconciler) syncChildren(log logr.Logger, inst *api.HierarchyConfiguration, ns *forest.Namespace) {
-	// Update the most recent list of children from the forest
+func (r *HierarchyConfigReconciler) syncChildren(log logr.Logger, inst *api.HierarchyConfiguration, ns *forest.Namespace, hnsnms []string) {
+	// Update the most recent list of children from the forest.
 	inst.Status.Children = ns.ChildNames()
 
-	// Currently when HNS reconciler is not enabled, the "SubnamespaceConflict" condition is cleared and
-	// updated when a parent namespace syncChildren().
-	// TODO report the "SubnamespaceConflict" in the HNS reconciliation. See issue:
-	//  https://github.com/kubernetes-sigs/multi-tenancy/issues/490
-	if !r.HNSReconcilerEnabled {
-		// We'll reset any of these conditions if they occur.
-		ns.ClearAllConditions(api.SubnamespaceConflict)
-	}
+	// We'll first clear SubnamespaceConflict condition and set it if any owned
+	// subnamespaces are missing or claimed by a different parent.
+	ns.ClearAllConditions(api.SubnamespaceConflict)
```

Comment:
Is `SubnamespaceConflict` still possible? The annotation is now the source-of-truth and K8s will ensure that there's no conflict there.

The conflict is now reported in the HNS object itself, not as a condition.

I think this lets us delete most of the rest of this function too, right?

Reply:
The following is implemented based on if the annotation or the parent of the owned namespace can be changed. I am not sure if it's the case (e.g. webhook is bypassed, etc.). We can discuss on it!

Reply:
As discussed offline, since we use owner annotation as source of truth and always set the owner as parent, this should not happen. Say a child changed its spec.parent to something else, the HCR will force the parent to be owner (if applicable).